R/pr.R

```diff
@@ -0,0 +1,556 @@
+  pr_handle(pr,
+            "GET",
+            path = path,
+            handler = handler,
+            preempt = preempt,
+            serializer = serializer,
+            endpoint = endpoint,
+            ...)
+}
+
+#' @rdname pr_handle
+#' @export
+pr_post <- function(pr,
+                    path,
+                    handler,
+                    preempt,
+                    serializer,
+                    endpoint,
+                    ...) {
+  pr_handle(pr,
+            "POST",
+            path = path,
+            handler = handler,
+            preempt = preempt,
+            serializer = serializer,
+            endpoint = endpoint,
+            ...)
+}
+
+#' @rdname pr_handle
+#' @export
+pr_put <- function(pr,
+                   path,
+                   handler,
+                   preempt,
+                   serializer,
+                   endpoint,
+                   ...) {
+  pr_handle(pr,
+            "PUT",
+            path = path,
+            handler = handler,
+            preempt = preempt,
+            serializer = serializer,
+            endpoint = endpoint,
+            ...)
+}
+
+#' @rdname pr_handle
+#' @export
+pr_delete <- function(pr,
+                      path,
+                      handler,
+                      preempt,
+                      serializer,
+                      endpoint,
+                      ...) {
+  pr_handle(pr,
+            "DELETE",
+            path = path,
+            handler = handler,
+            preempt = preempt,
+            serializer = serializer,
+            endpoint = endpoint,
+            ...)
+}
+
+#' @rdname pr_handle
+#' @export
+pr_head <- function(pr,
+                    path,
+                    handler,
+                    preempt,
+                    serializer,
+                    endpoint,
+                    ...) {
+  pr_handle(pr,
+            "HEAD",
+            path = path,
+            handler = handler,
+            preempt = preempt,
+            serializer = serializer,
+            endpoint = endpoint,
+            ...)
+}
+
+#' Mount a plumber router
+#'
+#' Plumber routers can be “nested” by mounting one into another
+#' using the `mount()` method. This allows you to compartmentalize your API
+#' by paths which is a great technique for decomposing large APIs into smaller
+#' files. This function mutates the plumber router (\code{pr}) in place, but
+#' also invisibly returns the updated router.
+#'
+#' @param pr the host plumber router.
+#' @param path a character string. Where to mount router.
+#' @param router a plumber router. Router to be mounted.
+#'
+#' @return a plumber router with the supplied router mounted
+#'
+#' @examples
+#' \dontrun{
+#' pr1 <- pr() %>%
+#'   pr_get("/hello", function() "Hello")
+#'
+#' pr() %>%
+#'   pr_get("/goodbye", function() "Goodbye") %>%
+#'   pr_mount("/hi", pr1) %>%
+#'   pr_run()
+#' }
+#'
+#' @export
+pr_mount <- function(pr,
+                     path,
+                     router) {
```

Comment:
Add a check like ``if (!inherits(pr, "plumber")) stop("`pr` must be an object of class 'plumber'.")``